lib/extensions/stroke_to_lpe_satin.py

```diff
@@ -148,13 +152,21 @@ def get_path(self, min_width, max_width, length, to_unit):
         el2.apply_transform()
         path2 = el2.get_path()
 
-        return str(path1) + str(path2)
+        # setup a rung
+        point1 = list(path1.end_points)[self.rung_node]
+        point2 = list(path2.end_points)[self.rung_node]
+
+        rungs = ''
+        if add_rungs:
+            rungs = f' M {point1[0]} {point1[1] + 0.1} L {point2[0]} {point2[1] - 0.2}'
+
+        return str(path1) + str(path2) + rungs
 
 
-satin_patterns = {'normal': SatinPattern('M 0,0.4 H 8', 'cc'),
+satin_patterns = {'normal': SatinPattern('M 0,0.4 H 4 H 8', 'cc'),
                   'pearl': SatinPattern('M 0,0 C 0,0.22 0.18,0.4 0.4,0.4 0.62,0.4 0.8,0.22 0.8,0', 'csc'),
                   'diamond': SatinPattern('M 0,0 0.4,0.2 0.8,0', 'ccc'),
-                  'triangle': SatinPattern('M 0.0,0 0.8,0.2 V 0', 'cccc'),
+                  'triangle': SatinPattern('M 0.0,0 0.4,0.1 0.8,0.2 V 0', 'cccc'),
```

Comment:
I tested with this and it got rid of the TooManyIntersections:

```
M 0.0,0 0.8,0.2 0.81,0
```

At least I think that's what I used.  Just turning that vertical line into a not-quite-vertical line.

Reply:
Ok, we will use that. I just add an extra node, so if we add rungs, it will not be at this straight line but in the middle.